sdk/textanalytics/test-resources.json

```diff
@@ -81,22 +78,37 @@
         },
         {
             "type": "Microsoft.Authorization/roleAssignments",
-            "apiVersion": "2018-09-01-preview",
-            "name": "[guid(resourceGroup().id)]",
+            "apiVersion": "[variables('authorizationApiVersion')]",
+            "name": "[guid(concat(variables('cognitiveServiceUserRoleId'), variables('uniqueSubDomainName')))]",
+            "dependsOn": [
+                "[variables('uniqueSubDomainName')]"
+            ],
             "properties": {
-                "roleDefinitionId": "[variables('roleDefinitionId')]",
+                "roleDefinitionId": "[variables('cognitiveServiceUserRoleId')]",
                 "principalId": "[parameters('testApplicationOid')]"
             }
         }
     ],
     "outputs": {
+        "AZURE_TENANT_ID": {
+            "type": "string",
+            "value": "[parameters('tenantId')]"
+        },
+        "AZURE_CLIENT_ID": {
+            "type": "string",
+            "value": "[parameters('testApplicationId')]"
+        },
+        "AZURE_CLIENT_SECRET": {
+            "type": "string",
+            "value": "[parameters('testApplicationSecret')]"
+        },
         "TEXT_ANALYTICS_API_KEY": {
             "type": "string",
-            "value": "[listKeys(resourceId('Microsoft.CognitiveServices/accounts',variables('uniqueSubDomainName')), '2017-04-18').key1]"
+            "value": "[listKeys(resourceId('Microsoft.CognitiveServices/accounts', variables('uniqueSubDomainName')), variables('cognitiveServicesApiVersion')).key1]"
         },
         "TEXT_ANALYTICS_ENDPOINT": {
             "type": "string",
             "value": "[variables('endpointValue')]"
         }
```

Comment:
Any automatically-added variables will actually use "TEXTANALYTICS" (no underscore) so if you changed these to follow, they will be consistent. In fact, at that point you don't even need the prefix since we look it up by default in `TestEnvironment` based on the service directory, "textanalytics". You could just define `API_KEY` and `ENDPOINT`. That said, I typically like the prefixes explicit because I tend to define long-lived apps as user env vars on my system. Just a tip.